File: Leetcode/1422. Maximum Score After Splitting a String.py
class Solution:
    def maxScore(self, s: str) -> int:
        n = len(s)
        score = 0
        zero = 0
        one = 0
        for i in range(n):
            if s[i] == '1':
                one += 1
        for i in range(n - 1):
            if s[i] == '0':
                zero += 1
            else:
                zero -= 1
            score = max(score,zero + one)
        return score
            
        
    

File: Leetcode/1422. Maximum Score After Splitting a String.cpp
class Solution {
public:
    int maxScore(string s) {
        int one = 0, n = s.length();
        for(int i = 0; i<n; i++) {
            if(s[i] != '0') one++;
        }
        int ans = INT_MIN, zero = 0;
        for(int i = 0; i<n - 1; i++) {
            if(s[i] != '0') one--;
            else zero++;
            ans = max(ans, one + zero);
        }
        return ans;
    }
};